src/schema.rs

```diff
@@ -272,24 +314,41 @@ impl Deserial for Fields {
     }
 }
 
-impl Serial for Module {
+impl Serial for ModuleV0 {
+    fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> {
+        self.contracts.serial(out)?;
+        Ok(())
+    }
+}
+
+impl Serial for ModuleV1 {
     fn serial<W: Write>(&self, out: &mut W) -> Result<(), W::Err> {
         self.contracts.serial(out)?;
         Ok(())
     }
 }
 
-impl Deserial for Module {
+impl Deserial for ModuleV0 {
     fn deserial<R: Read>(source: &mut R) -> ParseResult<Self> {
         let len: u32 = source.get()?;
         let contracts = deserial_map_no_length_no_order_check(source, len as usize)?;
```

Comment:
I think we should just use `deserial`, which does the exact same thing.

Reply:
I don't particularly want to change this. I think it is OK to be explicit here.